src/couch_replicator_manager.erl

```diff
@@ -232,14 +238,28 @@ handle_call(Msg, From, State) ->
     {stop, {error, {unexpected_call, Msg}}, State}.
 
 handle_cast({resume_scan, DbName}, State) ->
-    Since = case ets:lookup(?DB_TO_SEQ, DbName) of
-        [] -> 0;
-        [{DbName, EndSeq}] -> EndSeq
+    Pids = State#state.rep_start_pids,
+    NewPids = case lists:keyfind(DbName, 1, Pids) of
+        {DbName, _Pid} ->
+            Entry = case ets:lookup(?DB_TO_SEQ, DbName) of
+                [] ->
+                    {DbName, 0, true};
+                [{DbName, EndSeq, _Rescan}] ->
+                    {DbName, EndSeq, true}
+            end,
+            true = ets:insert(?DB_TO_SEQ, Entry),
+            Pids;
+        false ->
+            Since = case ets:lookup(?DB_TO_SEQ, DbName) of
+                [] -> 0;
+                [{DbName, EndSeq, _Rescan}] -> EndSeq
+            end,
+            true = ets:insert(?DB_TO_SEQ, {DbName, Since, false}),
+            Pid = start_changes_reader(DbName, Since),
+            couch_log:debug("Scanning ~s from update_seq ~p", [DbName, Since]),
+            [{DbName, Pid} | Pids]
     end,
-    ensure_rep_ddoc_exists(DbName),
```

Comment:
Basically, this means that I can remove _design/_replicator in any time and create malformed replication docs with any structure. Causing something like this:
```
{
  "_id": "ebb651e46435ca60243060583b000234",
  "_rev": "2-0f311c3a2590344c2c37f3e87b4f6ffd",
  "owner": null,
  "_replication_state": "error",
  "_replication_state_time": "2015-11-12T13:05:36+03:00",
  "_replication_state_reason": "{error,function_clause}"
 }
```

Or even worse - I can break replication status workflow in any way, everyone may edit every docs, everything is ruined. We shouldn't let this happened.